app/javascript/dashboard/routes/dashboard/settings/macros/MacroEditor.vue

```diff
@@ -1,9 +1,144 @@
 <template>
-  <div>MacrosEditor</div>
+  <div class="column content-box">
+    <p v-if="!uiFlags.isFetching && !macro" class="no-items-error-message">
+      {{ $t('MACROS.LIST.404') }}
+    </p>
+    <woot-loading-state
+      v-if="uiFlags.isFetching"
+      :message="$t('MACROS.LOADING')"
+    />
+    <macro-form v-if="macro" :macro-data.sync="macro" @submit="saveMacro" />
+  </div>
 </template>
 
 <script>
-export default {};
+import MacroForm from './MacroForm';
+import { MACRO_ACTION_TYPES } from './constants';
+import { mapGetters } from 'vuex';
+import { emptyMacro } from './macroHelper';
+import actionQueryGenerator from 'dashboard/helper/actionQueryGenerator.js';
+import alertMixin from 'shared/mixins/alertMixin';
+import macrosMixin from 'dashboard/mixins/macrosMixin';
+
+export default {
+  components: {
+    MacroForm,
+  },
+  mixins: [alertMixin, macrosMixin],
+  provide() {
+    return {
+      macroActionTypes: this.macroActionTypes,
+    };
+  },
+  data() {
+    return {
+      macro: {},
+      mode: 'CREATE',
+      macroActionTypes: MACRO_ACTION_TYPES,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      uiFlags: 'macros/getUIFlags',
+      labels: 'labels/getLabels',
+      teams: 'teams/getTeams',
+    }),
+    macroId() {
+      return this.$route.params.macroId;
+    },
+  },
+  watch: {
+    $route: {
+      handler() {
+        if (this.$route.params.macroId) {
+          this.fetchMacro();
+        } else {
+          this.initNewMacro();
+        }
+      },
+      immediate: true,
+    },
+  },
+  methods: {
+    fetchMacro() {
+      this.mode = 'EDIT';
+      this.$store.dispatch('agents/get');
+      this.$store.dispatch('teams/get');
+      this.$store.dispatch('labels/get');
+      this.manifestMacro();
+    },
+    async manifestMacro() {
+      const isMacroAvailable = this.$store.getters['macros/getMacro'](
+        this.macroId
+      );
+      if (isMacroAvailable) this.macro = this.formatMacro(isMacroAvailable);
+      else {
+        const unserializedMacro = await this.getSingleMacro();
+        this.macro = this.formatMacro(unserializedMacro);
+      }
+    },
```

Comment:
Please rewrite this as discussed over the call yesterady.

Reply:
@pranavrajs Updated this - the macro is now populated from the store while editing.